main.go

```diff
@@ -60,25 +60,20 @@ func main() {
 		printErrorFatal("Invalid Usage", err)
 	}
 
-	if len(ctx.Args) > 0 {
-		switch ctx.Args[0] {
-
-		case "version":
-			if Version == "" {
-				if info, ok := debug.ReadBuildInfo(); ok && info.Main.Sum != "" {
-					Version = info.Main.Version
-				} else {
-					Version = "unknown (built from source)"
-				}
-			}
-			version := fmt.Sprintf("freeze version %s", Version)
-			if len(CommitSHA) >= shaLen {
-				version += " (" + CommitSHA[:shaLen] + ")"
+	if config.Version {
+		if Version == "" {
+			if info, ok := debug.ReadBuildInfo(); ok && (info.Main.Version != "" && info.Main.Version != "(devel)") {
+				Version = info.Main.Version
+			} else {
+				Version = "latest (built from source)"
```

Comment:
I would have moved this line before the if to make it the default value, so there would be no else branch